src/runtime/resolver.rs

```diff
@@ -41,8 +42,14 @@ impl AsyncResolver {
         Ok(lookup)
     }
 
-    pub async fn txt_lookup<N: IntoName>(&self, query: N) -> Result<TxtLookup> {
-        let lookup = self.resolver.txt_lookup(query).await?;
-        Ok(lookup)
+    pub async fn txt_lookup<N: IntoName>(&self, query: N) -> Result<Option<TxtLookup>> {
```

Comment:
The only caller of this ignored the NoRecordsFound error, so I updated this to include that instead and just return an `Option`.